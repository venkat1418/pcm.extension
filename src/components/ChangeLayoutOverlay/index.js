import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import LayoutImages from './LayoutImages';
import LeftNavBtn from './LeftNavBtn';
import SaveCancelBtns from './SaveCancelBtns';
import RightNavBtn from './RightNavBtn';
import {changeLayoutServices} from '../../services/changeLayoutServices';

class ChangeLayoutOverlay extends Component {

    componentWillMount() {
      const recsPerPage = 8; // to be calculated baded on page size
      this.setState({
        pageNumber: 1,
        layoutSelected:"",
        layoutData: [],
        totalPageCount: 1,
        recsPerPage
      });
      changeLayoutServices.getLayoutData(this.props.zone).then(
          resp=>{
            const count = Math.ceil(resp.data/recsPerPage);
            this.setState({ layoutData: resp.data, totalPageCount: count });
          }
      );
    };

    changePage = (row) =>{
        this.setState({ pageNumber: row });
        const elements = document.getElementsByClassName("currentLayout");
        elements.forEach((element) => {element.classList.remove("currentLayout");})
        setTimeout(() => { this.layoutSelectionPageChange()} ,100 );
    }

    layoutSelectionPageChange = () => {
      const {pageNumber, recsPerPage, layoutData, layoutSelected } = this.state;

      const pageStartIndex = (pageNumber-1) *(recsPerPage);
      const pageEndIndex = (pageStartIndex + recsPerPage) < layoutData.length
          ? pageStartIndex + recsPerPage : layoutData.length;

      _.range(pageStartIndex, pageEndIndex).forEach((count) => {
        if(this.props.moduleId === layoutData[count].moduleId && layoutSelected !== "") {
            document.getElementById(`layoutSelect${this.props.moduleId}`).classList.remove("currentLayout");
        }
        else {
          document.getElementById(layoutData[count].moduleId).classList.add("layoutRowsNonSelect");
        }
        if(layoutSelected === `layoutSelect${layoutData[count].moduleId}`){
            document.getElementById(layoutSelected).classList.add("currentLayout");
            document.getElementById(layoutData[count].moduleId).classList.remove("layoutRowsNonSelect");
            document.getElementById(layoutData[count].moduleId).classList.add("layoutRows");
        }
      });
    }

    layoutIdSelected = (id) => {
        this.setState({ layoutSelected: id });
    }

    render() {
        if(this.props.showOverlay) {
            return (
              <div className='changeLayout'>
                <div id="cl_container" className='changeLayout_Container'>
                  <SaveCancelBtns
                    moduleIndex={this.props.moduleIndex}
                    currentModule={this.props.moduleId}
                    layoutData={this.state.layoutData}
                    changeLayoutClose={this.props.changeLayoutClose}
                    selected={this.state.layoutSelected}
                        />
                  <span className="changeLayoutSpan">CHANGE LAYOUT</span>
                  <LeftNavBtn pageNumber={this.state.pageNumber} changePage={this.changePage} />
                  <LayoutImages
                    moduleId={this.props.moduleId}
                    layoutData={this.state.layoutData}
                    pageNumber={this.state.pageNumber}
                    totalPageCount={this.state.totalPageCount}
                    layoutIdSelected={this.layoutIdSelected}
                    recsPerPage={this.state.recsPerPage} />
                  <RightNavBtn
                    pageNumber={this.state.pageNumber}
                    changePage={this.changePage}
                    totalPageCount={this.state.totalPageCount} />
                </div>
              </div>
            );
        }
        return null;
    }
};

ChangeLayoutOverlay.propTypes = {
  moduleId: PropTypes.string.isRequired,
  zone: PropTypes.string.isRequired,
  moduleIndex: PropTypes.number.isRequired,
  showOverlay: PropTypes.bool.isRequired,
  changeLayoutClose: PropTypes.func.isRequired
}

export default ChangeLayoutOverlay;
