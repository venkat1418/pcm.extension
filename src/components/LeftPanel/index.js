import React, { Component } from 'react';
import {DragDropContext} from 'react-beautiful-dnd';
import {NotificationManager} from 'react-notifications';
import PropTypes from 'prop-types';
import Module from './Module';
import {DNDHelper} from '../../helper/helper';
import messages from '../../messages/notification';


class LeftPanel extends Component {
    constructor(props){
        super(props);
        this.state={
            dndResult: {}
        }
    }
    onDragStart = ()=>{
      (document.getElementsByTagName("body")[0]).classList.add('body_no_scroll');
    }

    onDragEnd = (result)=>{
        (document.getElementsByTagName("body")[0]).classList.remove('body_no_scroll');
        if(!result.destination){
            return;
        }
        if (!result.draggableId ){
          return;
        }
        // let startStoryId = result.draggableId;
        const reorderData = DNDHelper.getEndStoryid(result, this.props.data);
        if (!reorderData) return;
        this.props.onReorder(reorderData);
        NotificationManager.success(messages.reOrder);
    }

    getModules = (moduleList) => {
        const rows = moduleList.map( (row, index) => (
          <Module
            key={index}
            name={row.name}
            zone={row.zone}
            content={row.blocks}
            source={row.source}
            sourceName={row.sourceName? row.sourceName:''}
            moduleId={row.moduleId}
            moduleIndex={index}
            details={row.details}
            editable={row.editable}
            display={row.display}
            proUser={row.proUser}
            />
          ));
        return rows;
    };

    render(){
        return (
          <div className="left_nav_panel">
            <DragDropContext onDragEnd={this.onDragEnd} onDragStart={this.onDragStart}>
              <div className="left_nav_body">
                {this.getModules(this.props.data)}
              </div>
            </DragDropContext>
          </div>
        );
    };
  }

LeftPanel.propTypes = {
  data: PropTypes.array.isRequired,
  onReorder: PropTypes.func.isRequired,
}

export default LeftPanel;
