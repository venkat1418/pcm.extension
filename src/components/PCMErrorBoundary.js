import React, { Component } from 'react';
// import {NotificationManager} from 'react-notifications';

export default class PCMErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });
    // You can also log the error to an error reporting service
    // logErrorToMyService(error, info);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      // NotificationManager.error(`Status is not 200!\n\nCurrent Status is ${resp.status}`)
      return <h1>Something went wrong. Please try again</h1>;
    }
    return this.props.children;
  }
}
