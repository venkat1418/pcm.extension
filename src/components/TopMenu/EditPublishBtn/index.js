// TODO: Need to refactor to move the logic to PCM server.

import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import { Loader } from 'react-overlay-loader';
import {NotificationManager} from 'react-notifications';
import 'react-overlay-loader/styles.css';
import PropTypes from 'prop-types';
import {PageStatusService} from '../../../services/PageStatusService';
import {AssetsStatusService} from '../../../services/AssetsStatusService';
import {toggleEditMode} from '../../../actions';
import {AuditLogService} from '../../../services/AuditLogService';
import errHandler from '../../../helper/errHandler';
import ConfirmDialogModal from '../../modals/ConfirmDialogModal';
import messages from '../../../messages/notification';
import popupMessages from '../../../messages/popupMessages';
import Loading from '../../modals/Loading';

export class EditPublishButton extends Component {

    constructor(props){
        super(props);
        this.onUnlockPage = this.onUnlockPage.bind(this);
        this.onUnlockAssets = this.onUnlockAssets.bind(this);
        this.state={
            assetsLockedBy: {},
            showAssetsStatusModal: false,
            pending: false,
            showCommonModal: false,
            showPublishModal: false
        }

    }

    openAssetsModal = ()=>{
        this.setState({ showAssetsStatusModal: true });
    }
    closeAssetsModal = ()=>{
        this.setState({ showAssetsStatusModal: false });
    }

    edit = () => {
        document.getElementById('pcmBody').style.marginLeft = "25%";
        document.getElementById('pcmBody').style.marginRight = "15%";
        if(this.state.pending) return false;
        this.setState({ pending: true });
            // AssetsStatusService.check(this.props.currentMetadata.cmsAuthToken, this.props.currentMetadata.assetsId, this.props.userData.uid).then(resp=>{
    AssetsStatusService.check(this.props.currentMetadata.cmsAuthToken, this.props.currentMetadata.assetsId, this.props.userData,this.props.currentMetadata.pageId).then(resp=>{
            this.setState({ pending: false });
            if(resp.status !== 200){
                NotificationManager.error(`Status is not 200!\n\nCurrent Status is ${resp.status}`)
                return false;
            }
            if(resp.data.error) {
                NotificationManager.error(`Error while checking asset status!`);
                return false;
            }
            if(resp.data.user.length>0){
               // this.onUnlockPage();
               this.setState({ pending: false });
               this.props.toggleEditMode(true);
               this.closeAssetsModal();
               document.getElementById("pcmBody").style.pointerEvents = "none";
            } else if(resp.data.concurrent.length>0){
                this.setState({ assetsLockedBy: resp.data.concurrent[0].user });
                this.openAssetsModal();
            }
        })
        .catch( err=>{ this.setState({ pending: false }); errHandler.notification(err); });
        return null;
    };

    onUnlockAssets(){
        this.setState({ pending: true });
        // AssetsStatusService.unlockThenLock(this.props.currentMetadata.cmsAuthToken, this.props.currentMetadata.assetsId, this.props.userData.uid).then(resp=>{
            AssetsStatusService.unlockThenLock(this.props.currentMetadata.cmsAuthToken, this.props.currentMetadata.assetsId, this.props.userData,this.props.currentMetadata.pageId).then(resp=>{
            this.setState({ pending: false });
            if(resp.status !== 200){
                NotificationManager.error(`Status is not 200!\n\nCurrent Status is ${resp.status}`)
                return false;
            }
            if(resp.data.error){
                NotificationManager.error(`Error while assets lock & unlock!`)
                return false;
            }
            if(resp.data.user.length>0){
               // this.onUnlockPage();
               this.setState({ pending: false });
               this.props.toggleEditMode(true);
               this.closeAssetsModal();
               document.getElementById("pcmBody").style.pointerEvents = "none";
            }
            else{
                NotificationManager.error(`Unlocking asset is failed!`)
            }
        })
        .catch( err=>{ this.setState({ pending: false }); errHandler.notification(err); });
    }

    onUnlockPage(){
        this.setState({ pending: true });
        PageStatusService.unlockPage(this.props.currentMetadata.pageId, this.props.userData).then(resp=>{
            this.setState({ pending: false });
            if(resp.data === 'yes'){
                const logData={
                    action: 'unlock_page',
                    pageId: this.props.currentMetadata.pageId,
                    user: this.props.userData,
                    updatedAt: new Date().getTime()
                }
                AuditLogService.logAuditLog(logData).then(resp=>{});
                this.setState({ pending: true });
                PageStatusService.lockPage(this.props.currentMetadata.pageId, this.props.userData).then(resp=>{
                    this.setState({ pending: false });
                    if(resp.data === 'yes'){
                        const logData={
                            action: 'lock_page',
                            pageId: this.props.currentMetadata.pageId,
                            user: this.props.userData,
                            updatedAt: new Date().getTime()
                        }
                        AuditLogService.logAuditLog(logData).then(resp=>{});
                        this.props.toggleEditMode(true);
                        this.closeAssetsModal();
                        document.getElementById("pcmBody").style.pointerEvents = "none";
                    }
                })
                .catch(err=>{ this.setState({ pending: false }); NotificationManager.error(`Error while locking page!`) })
            }
        })
        .catch(err=>{ this.setState({ pending: false }); NotificationManager.error(`Error while unlocking page!`) })
    }

    publish = () => {
        this.setState({ showPublishModal: true });
    };

    onPublish = () =>{
      if(this.state.pending) return false;
      this.setState({ pending: true, showPublishModal: false });
      Promise.all([
          PageStatusService.unlockPage(this.props.currentMetadata.pageId, this.props.userData),
          AssetsStatusService.removeLock(this.props.currentMetadata.cmsAuthToken, this.props.currentMetadata.assetsId, this.props.userData.uid)
      ]).then(resp=>{
          this.setState({ pending: false });
          if(resp[0].data !== 'yes'){
              NotificationManager.error(`Unlocking page is failed!`)
              return false;
          }
          if(resp[1].status !== 200){
              NotificationManager.error(`Status is not 200!\n\nCurrent Status is ${resp[1].status}!`)
              return false;
          }
          if(!resp[1].data){
              NotificationManager.error(`Unlocking asset is failed!`)
              return false;
          }
          const logData={
              action: 'unlock_page',
              pageId: this.props.currentMetadata.pageId,
              user: this.props.userData,
              updatedAt: new Date().getTime()
          }
          AuditLogService.logAuditLog(logData).then(resp=>{});
          this.props.toggleEditMode(false);
          NotificationManager.success(messages.publishMessage);
          document.getElementById("pcmBody").style.pointerEvents = "auto";
      })
      .catch(err=>{ this.setState({ pending: false }); })
    }

    onCancelPublish = () =>{
        this.setState({showPublishModal: false});
    }

    discard = (e)=>{
        if(this.state.pending) return false;
        if(this.props.isPageDirty) {
            e.preventDefault();
            this.setState({showCommonModal: true});
        } else {
            this.props.toggleEditMode(false);
            document.getElementById('pcmBody').style.marginLeft = '20%';
            document.getElementById('pcmBody').style.marginRight = '20%';
            NotificationManager.info(messages.discardButtonMsg_unlock);
        }
        return null;
    }

    onDiscard = (e) =>{
        this.setState({showCommonModal: false});
        this.props.toggleEditMode(false);
        NotificationManager.info(messages.discardButtonMsg_confirm);
        document.getElementById('pcmBody').style.marginLeft = '20%';
        document.getElementById('pcmBody').style.marginRight = '20%';
    }
    onCancelDiscard = () =>{
        this.setState({showCommonModal: false});
    }

    render() {
      const lockedBy = `${this.state.assetsLockedBy.name} is currently editing this page`;
        return (
          <div className="editPublishComp">
            {
                    /* <button className="edit_toggle_button" disabled={!this.props.isLogin} onClick={(this.props.isEditMode? this.discard:this.edit)}>{(this.props.isEditMode?"DISCARD":"EDIT")}</button> */

                    this.props.isEditMode?
                      <button className="edit_toggle_button" disabled={!this.props.isLogin} onClick={this.discard}>DISCARD</button>
                    :
                      <button className="edit_toggle_button" disabled={!this.props.isLogin} onClick={this.edit}>EDIT</button>

                }
            <button className="publishButton" disabled={!this.props.isEditMode || this.props.noOfOpenStories !== 0} onClick={this.publish}>PUBLISH</button>
            {/*
                    this.props.isEditMode?
                    <img alt="" className={"lock "+(this.props.isLogin ? "": "disabledItem")} height="40px" src="images/lock.png" onClick={this.props.isLogin? this.discard : ''} />
                    :
                    <img alt="" className={"lock "+(this.props.isLogin ? "": "disabledItem")} height="40px" src="images/unlock2.png" onClick={this.props.isLogin? this.edit : ''}/>
                */}
            <ConfirmDialogModal
              showModal={this.state.showAssetsStatusModal}
              mainDescription={popupMessages.editLocked.mainDescription}
              originalHeadline={lockedBy}
              onConfirm={this.onUnlockAssets}
              onCancel={this.closeAssetsModal}
              confirmLabel='UNLOCK'
                />
            <ConfirmDialogModal
              showModal={this.state.showCommonModal}
              mainDescription={popupMessages.discardPage.mainDescription}
              originalHeadline={popupMessages.discardPage.originalHeadline}
              onConfirm={this.onDiscard}
              onCancel={this.onCancelDiscard}
              confirmLabel='DISCARD'
                />
            <ConfirmDialogModal
              showModal={this.state.showPublishModal}
              mainDescription={popupMessages.publishPage.mainDescription}
              originalHeadline={popupMessages.publishPage.originalHeadline}
              onConfirm={this.onPublish}
              onCancel={this.onCancelPublish}
              confirmLabel='PUBLISH'
                />

                <Loading
                    showLoadingModal={this.state.pending}
                  />
          </div>
        );
    }
};

const mapStateToProps = (state)=>({
        isLogin: state.isLogin,
        isEditMode: state.isEditMode,
        userData: state.userData,
        currentMetadata: state.currentMetadata,
        isPageDirty: state.data.isPageDirty
    })
const mapDispatchToProps = (dispatch)=>bindActionCreators({
        toggleEditMode
    }, dispatch)

EditPublishButton.propTypes = {
  userData: PropTypes.object.isRequired,
  currentMetadata: PropTypes.object.isRequired,
  toggleEditMode: PropTypes.func.isRequired,
  isEditMode: PropTypes.bool.isRequired,
  isLogin: PropTypes.bool.isRequired,
  noOfOpenStories: PropTypes.number,
  isPageDirty: PropTypes.bool.isRequired
}
export default connect(mapStateToProps, mapDispatchToProps)(EditPublishButton);
