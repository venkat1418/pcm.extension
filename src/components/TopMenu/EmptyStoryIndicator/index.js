import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ContentStatusLabel extends Component {
  constructor(props){
      super(props);
      this.state={
      }
  }

    render() {

        const message = (this.props.noOfOpenStories > 0 && this.props.isEditMode) ?
          (<div className="horizontal_flex">There are <span className="currentEmptyCount">
            <span className="red_count">{this.props.noOfOpenStories}</span> empty positions</span> on this page</div>) :
          <div className="horizontal_flex">Status of the page</div>;
        return (
          <div className="statusLabelComp">
            <div className="contentStatusLabel">
              {message}
            </div>
          </div>
        );
    }
};


ContentStatusLabel.propTypes = {
  noOfOpenStories: PropTypes.number,
  isEditMode: PropTypes.bool.isRequired
}

export default ContentStatusLabel;
