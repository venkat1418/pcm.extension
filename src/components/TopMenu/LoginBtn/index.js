// TODO: Move Logout logic to PCM server

import React, { Component } from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import { withCookies } from 'react-cookie';
import { NotificationManager } from 'react-notifications';
import PropTypes from 'prop-types';
import {toggleLogin, toggleEditMode, updateUserData, updateAuthToken} from '../../../actions';
import SetCookieModal from '../../modals/SetCookieModal';
import {GetUserDataService} from '../../../services/GetUserDataService';
import {AuditLogService} from '../../../services/AuditLogService';
import UserAvatar from '../UserAvatar';
import errHandler from '../../../helper/errHandler';


export class LoginBtn extends Component {
    constructor(props){
        super(props);
        this.onLoginWithCookie = this.onLoginWithCookie.bind(this);
        this.state={
            showLoginModal: false
        }
    }

    openLoginModal = ()=>{
        this.setState({ showLoginModal: true });
    }

    closeLoginModal = ()=>{
        this.setState({ showLoginModal: false });
    }

    componentDidMount(){
        if(this.props.cookies.get('toolbeltCookie')){
            const cookie = this.props.cookies.get('toolbeltCookie');
            GetUserDataService.getUser(cookie).then(resp=>{
                if(resp.data.error){
                    NotificationManager.error(`Login failed!`)
                    return false;
                }
                else if(resp.data.uid > 0){
                    const userData = resp.data;
                    this.props.updateUserData(userData);
                    this.props.toggleLogin(true);
                    this.props.updateAuthToken({ newToken: cookie })
                }
            })
            .catch(err=>{ errHandler.notification(err); })
        }
    }

    onLoginWithCookie(cookie, expires){
        if(cookie.trim() === '' || expires.trim() === '') return false;
        GetUserDataService.getUser(cookie).then(resp=>{
            if(resp.data.error){
                NotificationManager.error(`Login failed!`)
                return false;
            }
            else if(resp.data.uid > 0) {
                const userData = resp.data;
                this.props.updateUserData(userData);
                this.props.toggleLogin(true);
                this.props.updateAuthToken({ newToken: cookie })
                const logData = {
                    action: 'user_login',
                    user: userData,
                    updatedAt: new Date().getTime()
                }
                AuditLogService.logAuditLog(logData).then(resp=>{});
                this.onSetCookie(cookie, expires);
                this.closeLoginModal();
            }
        })
        .catch(err=>{ errHandler.notification(err); })
        return false;
    }

    onSetCookie(cookie, expires){
        if(expires){
            this.props.cookies.set('toolbeltCookie', cookie, { path: '/', expires: new Date(expires) });
        }
        else{
            this.props.cookies.set('toolbeltCookie', cookie, { path: '/' });
        }
    }

    render() {
        return (
          <div className="loginComp">
            {
              this.props.isLogin?
                <UserAvatar />
              :
                <div>
                  <button className="loginButton" onClick={this.openLoginModal}>LOGIN</button>
                </div>
          }
            <SetCookieModal
              showModal={this.state.showLoginModal}
              onConfirm={this.onLoginWithCookie}
              onCancel={this.closeLoginModal}
            />
          </div>
        );
    }
};

const mapStateToProps = (state)=>({
        isLogin: state.isLogin,
        isEditMode: state.isEditMode
    })
const mapDispatchToProps = (dispatch)=>bindActionCreators({
        toggleLogin,
        toggleEditMode,
        updateUserData,
        updateAuthToken
    }, dispatch)

LoginBtn.propTypes = {
  cookies: PropTypes.object.isRequired,
  updateUserData: PropTypes.func.isRequired,
  toggleLogin: PropTypes.func.isRequired,
  updateAuthToken: PropTypes.func.isRequired,
  isLogin: PropTypes.bool.isRequired,
}
export default withCookies(connect(mapStateToProps, mapDispatchToProps)(LoginBtn));
