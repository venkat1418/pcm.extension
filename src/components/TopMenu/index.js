import React, { Component } from 'react';
import {bindActionCreators} from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LoginBtn from './LoginBtn';
import PageEditStatusLabel from './PageEditStatusLabel';
import EmptyStoryIndicator from './EmptyStoryIndicator';
import EditPublishBtn from './EditPublishBtn';
import ProUserToggle from './ProUserToggle';
import Preview from './Preview'
import LeftPanelContainer from '../../containers/LeftPanelContainer';
import { toggleLogin, toggleEditRole } from '../../actions';
import PCMErrorBoundary from '../PCMErrorBoundary'
import pino from '../../logger'

export class TopMenu extends Component {
    constructor(props){
        super(props);
        this.state={
        }
    }

    setEmptyStoryMesage = (noOfEmptyPositions) => {
        this.setState({noOfOpenStories: noOfEmptyPositions});
    }
    render() {
      pino.debug('in TOP MENU Render');
        const leftPanelComp = this.props.isEditMode? <LeftPanelContainer setMsg={this.setEmptyStoryMesage} /> : "";

        return(
          <div className="topLeftBanner">
            <div className="menu_container">
              <LoginBtn />
              <PageEditStatusLabel />
              <Preview />
              <ProUserToggle />
              <EmptyStoryIndicator noOfOpenStories={this.state.noOfOpenStories} isEditMode={this.props.isEditMode} />
              <EditPublishBtn noOfOpenStories={this.state.noOfOpenStories} />
            </div>

            <PCMErrorBoundary>
              {leftPanelComp}
            </PCMErrorBoundary>
          </div>
        );
    }
}

const mapStateToProps = (state)=>({
        isEditMode: state.isEditMode,
        isLogin: state.isLogin
    })
const mapDispatchToProps = (dispatch)=>bindActionCreators({
        toggleLogin,
        toggleEditRole
    }, dispatch)

TopMenu.propTypes = {
  isEditMode: PropTypes.bool.isRequired
}
export default connect(mapStateToProps, mapDispatchToProps)(TopMenu);
