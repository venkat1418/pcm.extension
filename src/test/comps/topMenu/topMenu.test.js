import { shallow } from 'enzyme';
import React from 'react';
import {TopMenu} from "../../../components/TopMenu";

test('Test the renders of TopMenu', () => {
   const wrapper = shallow(<TopMenu />);
   //Create a snapshot with the default render of the component
   expect(wrapper).toMatchSnapshot();

   // set the state explicitly and create a snapshot again to check the changes with updated state
   wrapper.setState({noOfOpenStories:5});
   expect(wrapper).toMatchSnapshot();

  // This method is called from LeftPanelContainer .
   // Call the asynchrnous method and observe the changes in state
   wrapper.instance().setEmptyStoryMesage(4);
   expect(wrapper.state().noOfOpenStories).toBe(4);
});
