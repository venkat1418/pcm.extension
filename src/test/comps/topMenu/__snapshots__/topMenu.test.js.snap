// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the renders of TopMenu 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="menu_container"
        >
          <withCookies(Connect(LoginBtn)) />
          <Connect(PageNameLabel) />
          <ContentStatusLabel
            isEditMode={undefined}
            noOfOpenStories={undefined}
          />
          <Connect(EditPublishButton)
            noOfOpenStories={undefined}
          />
        </div>,
        <PCMErrorBoundary>
          
        </PCMErrorBoundary>,
      ],
      "className": "topLeftBanner",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withCookies(Connect(LoginBtn)) />,
            <Connect(PageNameLabel) />,
            <ContentStatusLabel
              isEditMode={undefined}
              noOfOpenStories={undefined}
            />,
            <Connect(EditPublishButton)
              noOfOpenStories={undefined}
            />,
          ],
          "className": "menu_container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isEditMode": undefined,
              "noOfOpenStories": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "noOfOpenStories": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="menu_container"
          >
            <withCookies(Connect(LoginBtn)) />
            <Connect(PageNameLabel) />
            <ContentStatusLabel
              isEditMode={undefined}
              noOfOpenStories={undefined}
            />
            <Connect(EditPublishButton)
              noOfOpenStories={undefined}
            />
          </div>,
          <PCMErrorBoundary>
            
          </PCMErrorBoundary>,
        ],
        "className": "topLeftBanner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withCookies(Connect(LoginBtn)) />,
              <Connect(PageNameLabel) />,
              <ContentStatusLabel
                isEditMode={undefined}
                noOfOpenStories={undefined}
              />,
              <Connect(EditPublishButton)
                noOfOpenStories={undefined}
              />,
            ],
            "className": "menu_container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isEditMode": undefined,
                "noOfOpenStories": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "noOfOpenStories": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test the renders of TopMenu 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="menu_container"
        >
          <withCookies(Connect(LoginBtn)) />
          <Connect(PageNameLabel) />
          <ContentStatusLabel
            isEditMode={undefined}
            noOfOpenStories={5}
          />
          <Connect(EditPublishButton)
            noOfOpenStories={5}
          />
        </div>,
        <PCMErrorBoundary>
          
        </PCMErrorBoundary>,
      ],
      "className": "topLeftBanner",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withCookies(Connect(LoginBtn)) />,
            <Connect(PageNameLabel) />,
            <ContentStatusLabel
              isEditMode={undefined}
              noOfOpenStories={5}
            />,
            <Connect(EditPublishButton)
              noOfOpenStories={5}
            />,
          ],
          "className": "menu_container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isEditMode": undefined,
              "noOfOpenStories": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "noOfOpenStories": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="menu_container"
          >
            <withCookies(Connect(LoginBtn)) />
            <Connect(PageNameLabel) />
            <ContentStatusLabel
              isEditMode={undefined}
              noOfOpenStories={5}
            />
            <Connect(EditPublishButton)
              noOfOpenStories={5}
            />
          </div>,
          <PCMErrorBoundary>
            
          </PCMErrorBoundary>,
        ],
        "className": "topLeftBanner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withCookies(Connect(LoginBtn)) />,
              <Connect(PageNameLabel) />,
              <ContentStatusLabel
                isEditMode={undefined}
                noOfOpenStories={5}
              />,
              <Connect(EditPublishButton)
                noOfOpenStories={5}
              />,
            ],
            "className": "menu_container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isEditMode": undefined,
                "noOfOpenStories": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "noOfOpenStories": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
